### PR TITLE
Changes to ManagedFields is not mutation for GC

### PR DESCRIPTION
**Pull request by apelisse**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
Fixed a regression preventing garbage collection of RBAC role and binding objects
```

### CODE REVIEW

**Comment by apelisse** [comment]
it's not super clear how we can know which new fields might want to be ignored? Should I just fail if any new field is added to `ObjectMeta`? @liggitt 

cc @deads2k

**Comment by liggitt** [comment]
> Should I just fail if any new field is added to `ObjectMeta`?

That's a reasonable place to start, along with a comment about what should be used to decide if new fields should be cleared or added to the list to ignore in the test

File: pkg/registry/rbac/helpers.go [hunk 1]
```diff
@@ -44,6 +44,7 @@ func IsOnlyMutatingGCFields(obj, old runtime.Object, equalities conversion.Equal
 	copiedMeta.SetOwnerReferences(oldMeta.GetOwnerReferences())
 	copiedMeta.SetFinalizers(oldMeta.GetFinalizers())
 	copiedMeta.SetSelfLink(oldMeta.GetSelfLink())
+	copiedMeta.SetManagedFields(oldMeta.GetManagedFields())
```
Comment (deads2k) [inline pkg/registry/rbac/helpers.go:47]: Seems like we want to deny taking ownership of fields outside of the GC fields, right?

Comment (apelisse) [reply pkg/registry/rbac/helpers.go:47]: yes, but it's fairly hard to inspect the managedFields to make this kind of validation.

**Comment by apelisse** [comment]
@liggitt PTAL

File: pkg/registry/rbac/helpers.go (hunk 1 above)
Comment (liggitt) [reply pkg/registry/rbac/helpers.go:47]: Keeping this check coarse-grained at the managedFields level seems ok to me, but TestIsOnlyMutatingGCFields needs updating to prove this is effective.

**Comment by liggitt** [comment]
/kind regression

this will need picking to 1.16 and 1.17 as well, right?

**Comment by apelisse** [comment]
Correct

File: pkg/registry/rbac/helpers.go (hunk 1 above)
Comment (liggitt) [reply pkg/registry/rbac/helpers.go:47]: oops, missed there was already a test

**Comment by liggitt** [comment]
Thanks. this fix needs to be picked back to existing releases.

**Pull request merged**